pkg/controller/installation/products/threescale/reconciler.go

```diff
@@ -430,6 +443,78 @@ func (r *Reconciler) reconcileUpdatingAdminDetails(ctx context.Context, serverCl
 	return v1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) reconcileOpenshiftUsers(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	logrus.Info("Reconciling openshift users to 3scale")
+
+	rhssoConfig, err := r.ConfigManager.ReadRHSSO()
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	kcr := &aerogearv1.KeycloakRealm{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: rhssoConfig.GetRealm(), Namespace: rhssoConfig.GetNamespace()}, kcr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	accessToken, err := r.GetAdminToken(ctx, serverClient)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	tsUsers, err := r.tsClient.GetUsers(*accessToken)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	added, deleted := r.getUserDiff(kcr.Spec.Users, tsUsers.Users)
+	for _, kcUser := range added {
+		if kcUser.UserName == rhsso.CustomerAdminUser.UserName {
+			continue
+		}
+
+		res, err := r.tsClient.AddUser(kcUser.UserName, kcUser.Email, "", *accessToken)
+		if err != nil || res.StatusCode != http.StatusCreated && res.StatusCode != http.StatusUnprocessableEntity {
+			return v1alpha1.PhaseFailed, err
+		}
+	}
+	for _, tsUser := range deleted {
+		res, err := r.tsClient.DeleteUser(tsUser.UserDetails.Id, *accessToken)
+		if err != nil || res.StatusCode != http.StatusOK {
+			return v1alpha1.PhaseFailed, err
+		}
+	}
+
+	openshiftAdminGroup := &usersv1.Group{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: "dedicated-admins"}, openshiftAdminGroup)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return v1alpha1.PhaseFailed, err
+	}
+	newTsUsers, err := r.tsClient.GetUsers(*accessToken)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+	for _, tsUser := range newTsUsers.Users {
+		if tsUser.UserDetails.Username == rhsso.CustomerAdminUser.UserName {
+			continue
+		}
+
+		if userIsOpenshiftAdmin(tsUser, openshiftAdminGroup) {
+			res, err := r.tsClient.SetUserAsAdmin(tsUser.UserDetails.Id, *accessToken)
+			if err != nil || res.StatusCode != http.StatusOK {
+				return v1alpha1.PhaseFailed, err
+			}
+		} else {
+			res, err := r.tsClient.SetUserAsMember(tsUser.UserDetails.Id, *accessToken)
+			if err != nil || res.StatusCode != http.StatusOK {
+				return v1alpha1.PhaseFailed, err
+			}
+		}
```

Comment:
The code above will "spam" http requests on each reconcile. Can we check tsUser.UserDetails.Role to see if it really needs to be updated?